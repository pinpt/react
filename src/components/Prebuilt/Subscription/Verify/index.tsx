import { ReactElement } from 'react';
import { IHeaderProps } from '../../../Header';
import Header from '../../Header';
import Footer from '../../Footer';
import VerifyComponent from '../../../Subscription/Verify';
import { ISite } from '../../../../lib/types/site';
import { ILogoProps } from '../../../Logo';
import { IThemeToggleProps } from '../../../ThemeToggle';
import withWrapper from '../../../Internal/withWrapper';
import { IFooterProps } from '../../../Footer';
import { ISocialMediaBarProps } from '../../../SocialMedia/SocialMediaBar';
import { ICopyrightProps } from '../../../Copyright';

export interface IPrebuiltVerifyProps {
	className?: string;
	site: ISite;
	renderHeader?: (site: ISite) => ReactElement<IHeaderProps>;
	handleSelectHome?: () => void;
	renderLogo?: (site: ISite) => ReactElement<ILogoProps>;
	renderThemeToggle?: (site: ISite) => ReactElement<IThemeToggleProps>;
	renderFooter?: (site: ISite) => ReactElement<IFooterProps>;
	renderSocial?: (site: ISite) => ReactElement<ISocialMediaBarProps>;
	renderCopyright?: (site: ISite) => ReactElement<ICopyrightProps>;
}

const Verify = (props: IPrebuiltVerifyProps) => {
	const {
		className = '',
		renderHeader,
		site,
		handleSelectHome,
		renderLogo,
		renderThemeToggle,
		renderFooter,
		renderCopyright,
		renderSocial,
	} = props;

	return (
		<div className={`Pinpoint Page SubscriptionVerify Prebuilt ${className}`}>
			{withWrapper(
				renderHeader?.(site) ?? (
					<Header
						site={site}
						handleSelectHome={handleSelectHome}
						renderLogo={renderLogo}
						renderSearch={() => <></>}
						renderSubscribe={() => <></>}
						renderThemeToggle={renderThemeToggle}
					/>
				),
				'header'
			)}
			{withWrapper(<VerifyComponent />, 'subscribe')}
			{withWrapper(
				renderFooter?.(site) ?? (
					<Footer
						site={site}
						renderCopyright={renderCopyright}
						renderLogo={renderLogo}
						renderSocial={renderSocial}
						renderSubscribe={() => <></>}
					/>
				),
				'footer'
			)}
		</div>
	);
};

export default Verify;
