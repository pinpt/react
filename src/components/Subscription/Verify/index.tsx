import React, { ChangeEvent, useCallback, useEffect, useMemo, useState } from 'react';
import useEmailAction from '../../../lib/hooks/useEmailAction';
import EmailAction from '../../EmailAction';
import Field from '../../Form/field';
import Form from '../../Form/form';
import Loader from '../../Loader';

export interface IVerifyeProps {
	className?: string;
	verified?: boolean;
	loading?: boolean;
	firstName?: string;
	lastName?: string;
	onSave?: (firstName: string, lastName: string) => Promise<any>;
	pending?: boolean;
}

const baseClass = 'Pinpoint SubscriptionVerify';

const Verify = (props: IVerifyeProps) => {
	const {
		className = '',
		verified = false,
		loading = false,
		firstName: propsFirstName = '',
		lastName: propsLastName = '',
		onSave,
		pending = false,
	} = props;
	const emailActionState = useEmailAction();
	const [firstName, setFirstName] = useState(() => propsFirstName);
	const [lastName, setLastName] = useState(() => propsLastName);

	useEffect(() => {
		if (verified) {
			emailActionState.setters.setError('');
			emailActionState.setters.setCritical(false);
			emailActionState.setters.setCanClear(false);
			emailActionState.setters.setMessage('Your email address has been verified');
		} else {
			emailActionState.setters.setCritical(true);
			emailActionState.setters.setCode(500);
			emailActionState.setters.setError('Your email could not be verified at this time');
			emailActionState.setters.setMessage('');
		}
	}, [verified]);

	const handleFirstNameChange = useCallback((e: ChangeEvent<HTMLInputElement>) => {
		setFirstName(e.target.value);
	}, []);

	const handleLastNameChange = useCallback((e: ChangeEvent<HTMLInputElement>) => {
		setLastName(e.target.value);
	}, []);

	const isFormDirty = useMemo(() => {
		return propsFirstName !== firstName || propsLastName !== lastName;
	}, [propsFirstName, propsLastName, firstName, lastName]);

	const handleSave = useCallback(() => {
		onSave?.(firstName, lastName);
	}, [firstName, lastName, onSave]);

	if (loading) {
		return <Loader />;
	}

	return (
		<div className={`${baseClass} Wrapper ${className}`}>
			<div className={`${baseClass} Content`}>
				<div className={`${baseClass} Inner`}>
					<EmailAction {...emailActionState} />
					{verified && (
						<div>
							<Form
								title="ðŸ‘‹ We'd love to know a little more about you!"
								description="If you prefer to, you can simply close this tab instead."
								onSave={handleSave}
								dirty={isFormDirty}
							>
								<div className={`${baseClass} Form`}>
									<Field label="First Name">
										<input
											value={firstName}
											className="Pinpoint SubscriptionSubscribe Field"
											onChange={handleFirstNameChange}
										/>
									</Field>
									<Field label="Last Name">
										<input
											value={lastName}
											className="Pinpoint SubscriptionSubscribe Field"
											onChange={handleLastNameChange}
										/>
									</Field>
								</div>
							</Form>
						</div>
					)}
				</div>
			</div>
		</div>
	);
};

export default Verify;
