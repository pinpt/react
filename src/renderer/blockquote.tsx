import { registerNode, recurseIntoChildren, NodeProps } from './register';

const BlockQuote = ({ node }: NodeProps) => {
	return <blockquote>{recurseIntoChildren(node)}</blockquote>;
};

registerNode('blockquote', (node) => <BlockQuote node={node} />);
