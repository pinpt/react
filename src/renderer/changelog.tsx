import { ReactYouTubeLite as Youtube } from 'react-youtube-lite';
import { getFileUrl } from '../lib/file_metadata';
import { slugifyContent } from '../lib/string';
import { ChangelogCoverMedia, ChangelogMediaType } from '../lib/types';
import { Content as Doc } from './';

const ImageMedia = ({ src, title = '' }: { src: string; title?: string }) => {
	return (
		<div className="image">
			<img src={src} alt={title} />
		</div>
	);
};

const VideoMedia = ({ src }: { src: string }) => {
	return (
		<div className="video">
			<video controls src={src} />
		</div>
	);
};

const YoutubeMediaSSR = ({ id, poster = 'hqdefault' }: { id: string; poster?: string }) => {
	// the component isn't SSR friendly so this is basically used to sort of mimic it for generated code
	const posterUrl = `https://i.ytimg.com/vi/${id}/${poster}.jpg`;
	return (
		<>
			{poster && <link rel="preload" href={posterUrl} as="image" />}
			<div
				className="youtube ryt-lite embed-responsive aspect-ratio-16/9"
				style={{ backgroundImage: `url(${posterUrl})` }}
			>
				<div className="lty-playbtn" data-url={`https://www.youtube-nocookie.com/embed/${id}?autoplay=1`} />
			</div>
		</>
	);
};

const YoutubeMedia = ({
	id,
	metadata,
	staticMode,
}: {
	id: string;
	metadata?: Record<string, any>;
	staticMode?: true;
}) => {
	return (
		<div className="youtube">
			{!staticMode ? (
				<Youtube url={`https://www.youtube.com/watch?v=${id}`} poster={metadata?.poster} />
			) : (
				<YoutubeMediaSSR id={id} poster={metadata?.poster} />
			)}
		</div>
	);
};

const Media = ({ media, title, staticMode }: { media?: ChangelogCoverMedia; title?: string; staticMode?: true }) => {
	if (!media) {
		return <></>;
	}
	let content: React.ReactElement;
	switch (media.type) {
		case ChangelogMediaType.None: {
			return <></>;
		}
		case ChangelogMediaType.Image: {
			content = <ImageMedia src={getFileUrl(media.value)} title={title} />;
			break;
		}
		case ChangelogMediaType.Video: {
			content = <VideoMedia src={getFileUrl(media.value)} />;
			break;
		}
		case ChangelogMediaType.Youtube: {
			content = <YoutubeMedia id={media.value} metadata={media.metadata} staticMode={staticMode} />;
			break;
		}
	}
	return <div className="media-container">{content}</div>;
};

interface ChangelogProps {
	id: string;
	title?: string;
	content?: any;
	coverMedia?: ChangelogCoverMedia;
	limit?: number;
	divider?: boolean;
	staticMode?: true;
}

const ChangelogLog = (props: ChangelogProps) => {
	return (
		<article className="changelog notebook-editor read-only">
			{props.coverMedia && (
				<section className="covermedia">
					<Media media={props.coverMedia} title={props.title} staticMode={props.staticMode} />
				</section>
			)}
			<section className="ProseMirror">
				<Doc node={props.content} limit={props.limit} />
				{props.limit && props.content.content?.length > props.limit && (
					<div className="continue">
						<a href={slugifyContent(props.id, props.title)}>Continue Reading â†’</a>
					</div>
				)}
			</section>
			{props.divider && <hr className="divider" />}
		</article>
	);
};

export default ChangelogLog;
