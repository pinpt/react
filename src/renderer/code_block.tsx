import { Prism as SyntaxHighlighter } from 'react-syntax-highlighter';
import { twilight } from 'react-syntax-highlighter/dist/cjs/styles/prism';
import { NodeProps, registerNode } from './register';

const CodeBlock = ({ node }: NodeProps) => {
	return (
		<SyntaxHighlighter
			language={node.attrs.language}
			style={twilight}
			customStyle={{
				background: 'black',
				fontFamily: 'ui-monospace',
				borderWidth: '0px',
				borderRadius: '.25em',
			}}
		>
			{node.content?.[0].text ?? ''}
		</SyntaxHighlighter>
	);
};

registerNode('code_block', (node) => <CodeBlock node={node} />);
