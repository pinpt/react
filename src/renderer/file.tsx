import {
	faDownload, faFile, faFileAlt, faFileArchive, faFileAudio, faFileCode, faFileExcel, faFilePdf,
	faFilePowerpoint, faFileVideo, faFileWord
} from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { NodeProps, registerNode } from './register';

export const iconForType = (type = '') => {
	if (/pdf/.test(type)) {
		return faFilePdf;
	}
	if (/zip|tar/.test(type)) {
		return faFileArchive;
	}
	if (/video/.test(type)) {
		return faFileVideo;
	}
	if (/audio/.test(type)) {
		return faFileAudio;
	}
	if (/excel|spreadsheet/.test(type)) {
		return faFileExcel;
	}
	if (/powerpoint/.test(type)) {
		return faFilePowerpoint;
	}
	if (/word/.test(type)) {
		return faFileWord;
	}
	if (/css|csv|graphql|html|java|json|php|python|typescript|xml/.test(type)) {
		return faFileCode;
	}
	if (/text/.test(type)) {
		return faFileAlt;
	}
	return faFile;
};

export const formattedSize = (size = 0) => {
	const kB = size / 1000;
	if (kB < 10) {
		return `${Math.round(kB * 10) / 10} KB`;
	}
	if (kB < 1000) {
		return `${Math.round(kB)} KB`;
	}
	const MB = kB / 1000;
	return `${Math.round(MB * 10) / 10} MB`;
};

const File = ({ node }: NodeProps) => {
	const { description, filename, publicUrl, size, type } = node.attrs;

	const onDownload = () => {
		if (publicUrl) {
			const u = new URL(publicUrl);
			u.searchParams.set('filename', filename);
			u.searchParams.set('disposition', 'attachment');
			const a = document.createElement('a');
			a.style.display = 'none';
			a.href = u.toString();
			// the filename you want
			a.download = filename;
			document.body.appendChild(a);
			a.click();
			a.remove();
		}
	};

	return (
		<div className="file">
			<div className="container">
				<FontAwesomeIcon icon={iconForType(type)} />
				<div className="details">
					{publicUrl ? (
						<a href={publicUrl} target="_blank" rel="noopener noreferrer" className="filename">
							{filename}
						</a>
					) : (
						<div className="filename">{filename}</div>
					)}
					{description && <div className="description">{description}</div>}
					<div className="size">{formattedSize(size)}</div>
				</div>
				<div className="options">
					{publicUrl && (
						<div className="option group" onClick={onDownload}>
							<FontAwesomeIcon icon={faDownload} />
							<span>Download</span>
						</div>
					)}
				</div>
			</div>
		</div>
	);
};

registerNode('file', (node) => <File node={node} />);
