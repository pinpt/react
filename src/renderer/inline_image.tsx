import { extractImageMetadataFromFileID } from '../lib/file_metadata';
import { NodeProps, registerNode } from './register';

const InlineImage = ({ node }: NodeProps) => {
	const { size } = extractImageMetadataFromFileID(node.attrs?.src);
	const width = size?.width;
	const height = size?.height;

	const image = (
		<img src={node.attrs?.src} width={width} height={height} alt={node.attrs?.alt} className="inline_image" />
	);

	return (
		<div>
			{node.attrs?.href ? (
				<a href={node.attrs.href} target="_blank" rel="noopener noreferrer">
					{image}
				</a>
			) : (
				<>{image}</>
			)}
		</div>
	);
};

registerNode('inline-image', (node) => <InlineImage node={node} />);
