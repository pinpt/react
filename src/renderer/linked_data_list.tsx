import { NodeProps, recurseIntoChildren, registerNode } from './register';

const LinkedDataList = ({ node }: NodeProps) => {
	return <div className="linked-data-list">{recurseIntoChildren(node)}</div>;
};

registerNode('linked_data_list', (node) => <LinkedDataList node={node} />);
