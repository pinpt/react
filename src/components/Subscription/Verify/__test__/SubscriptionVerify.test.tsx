import renderer from 'react-test-renderer';
import Verify from '..';

test('Test default state', () => {
	const component = renderer.create(<Verify verified />);
	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('Test not verified state', () => {
	const component = renderer.create(<Verify />);
	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
