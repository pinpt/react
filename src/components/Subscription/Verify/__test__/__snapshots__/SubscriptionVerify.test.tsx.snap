// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test default state 1`] = `
<div
  className="Pinpoint SubscriptionVerify Wrapper "
>
  <div
    className="Pinpoint SubscriptionVerify Content"
  >
    <div
      className="Pinpoint SubscriptionVerify Inner"
    >
      
      
      
      <div>
        <div
          className="Pinpoint Form Wrapper "
        >
          <div
            className="Pinpoint Form Parent "
          >
            <div
              className="Pinpoint Form Control"
            >
              <div
                className="Pinpoint Form Heading "
              >
                <h1>
                  ðŸ‘‹ We'd love to know a little more about you!
                </h1>
                <h2>
                  If you prefer to, you can simply close this tab instead.
                </h2>
              </div>
              <div>
                <button
                  className="Pinpoint SubscriptionVerify Save"
                  disabled={true}
                  onClick={[Function]}
                >
                  Save
                </button>
              </div>
            </div>
            <div
              className="Pinpoint SubscriptionVerify Form"
            >
              <div
                className="Pinpoint Field Wrapper "
              >
                <div
                  className="Pinpoint Field Parent"
                >
                  <label
                    className="Pinpoint Field Label "
                  >
                    First Name
                  </label>
                  <div
                    className="Pinpoint Field Status"
                  />
                </div>
                <div
                  className="Pinpoint Field Children"
                >
                  <input
                    className="Pinpoint SubscriptionSubscribe Field"
                    onChange={[Function]}
                    value=""
                  />
                </div>
                <div
                  className="Pinpoint Field Description"
                />
              </div>
              <div
                className="Pinpoint Field Wrapper "
              >
                <div
                  className="Pinpoint Field Parent"
                >
                  <label
                    className="Pinpoint Field Label "
                  >
                    Last Name
                  </label>
                  <div
                    className="Pinpoint Field Status"
                  />
                </div>
                <div
                  className="Pinpoint Field Children"
                >
                  <input
                    className="Pinpoint SubscriptionSubscribe Field"
                    onChange={[Function]}
                    value=""
                  />
                </div>
                <div
                  className="Pinpoint Field Description"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Test not verified state 1`] = `
<div
  className="Pinpoint SubscriptionVerify Wrapper "
>
  <div
    className="Pinpoint SubscriptionVerify Content"
  >
    <div
      className="Pinpoint SubscriptionVerify Inner"
    >
      <div
        className="Pinpoint Error "
      >
        <div
          className="logo"
        />
        <div
          className="error"
        >
          Your email could not be verified at this time
        </div>
        <h1
          className="title"
        >
          Something Went Wrong
        </h1>
        <div
          className="description"
        >
          Sorry, an unexpected error occurred.
        </div>
        <div
          className="link"
        />
      </div>
      
    </div>
  </div>
</div>
`;
